### ISSUE TITLE
Log Spam

### ISSUE THREAD

**Issue by abrachoo** [Ivorforce/Psychedelicraft#25]
This mod is spamming errors in the minecraft log.
![glerror](https://cloud.githubusercontent.com/assets/7889193/12407657/0391e37e-be0e-11e5-9fcb-f7f63d702575.png)

**Comment by XBlackLeopardX** [comment]
its an issue between this mod and botania i mentioned it about a year ago i believe. i think its on botanias end but not sure

**Comment by abrachoo** [comment]
I don't use botania.

**Comment by XBlackLeopardX** [comment]
in the config there is a shader option i think try and turn that off and see if it fixes the issue

**Comment by GunArm** [comment]
You mean turn off the shaders in psychedelicraft?  Isn't that sortof the entire point of the mod?

I also have this spam, and I don't have botania.  I remember reading about this in the past and Ivorforce seemed to say it was caused by having another mod which uses shaders, as he had not written this mod in a way that played well with any other mods having shaders, and didn't expect it would be used with other mods that have shaders.  However lots of mods seem to use shaders here and there for little things (botania is just one of them), and those mods don't seem to have this problem with each other, perhaps because they just don't log like this one.  

I think I remember him saying these error messages are "mostly harmless", except they make it difficult to read the logs, cause heavy disk IO, and greatly concern anyone who didn't make the pack and research it enough to have been told it's "not a big deal".

If the underlying problem cannot/will not be fixed, could we atleast mitigate the spam somehow?

**Comment by Ivorforce** [comment]
Post Render Pass errors are usually not _caused_ by Psychedelicraft - just logged (Since it wraps all rendering). I don't have time to look at this one specifically right now, since I don't actively work on Psychedelicraft. I'm keeping the issue open until I do look at it or it fixes itself from the other end.

Also, it's not that I didn't make Psychedelicraft with other shader mods in mind, it's that it's _impossible_ to make different shaders compatible. It's dumb to place your custom shader in place of the default render pipe (or running shader) for just one object if you don't own the global shaders - but I can hardly control that.

**Comment by GunArm** [comment]
Ok I do get what you are saying about the different shaders being incompatible and how the logging wraps all rendering.  

But, since the sheer volume of the logging itself is causing problems, could it just be disabled (even by config)?  Especially if it's not all coming from psychedelicraft anyway.  For example, I actually have no visual problems, but 150 meg log files, and a bunch of people trying to blame random unrelated problems on psychedelicraft just because it's spewing messages.

**Comment by Ivorforce** [comment]
Thing is, if the issue is an incompatibility, visual or not, it's important these get logged to be able to get to the bottom of this somehow.
If it isn't, disabling Psychedelicraft from logging will just propagate the problem to the next logger, so no ups here either.

If you want to see what's causing the issue, there's an OGL logging option in the psychedelicraftcore config file (debugGlErrorTraceDumps). If you enable this, you can see exactly what mod is the actual culprit and try to approach its author.

**Comment by GunArm** [comment]
Thanks.  I'll try that when i have time and report back.  I suspect certain TEs from both EnderIO and Draconic Evolution.

**Comment by Ivorforce** [comment]
I believe some EnderIO pipes use shaders, so that's very plausible :)

**Comment by GunArm** [comment]
I am not able to find that config file. :/  Is psychedelicraftcore part of the Psychedelicraft-1.5.2.jar?  I only see the regular psychedelicraft.cfg (which doesn't have that setting).  I do see psychedelicraft core in the "mod options" gui config, but the configure button is greyed out.

**Comment by Ivorforce** [comment]
I'm not completely sure if it's in the latest release. I may have added it in a nightly (which for Psychedelicraft are stable right now).

**Comment by GunArm** [comment]
The log was about 500 meg after about 2 minutes, I'm not sure how to share it other than zipping it up.

You'll have to check me because I'm not sure I'm reading it right.  But it seems like Reika's mods, Journeymaps, EnderIO, Draconic Evolution, and a few from psychedelicraft.

Now that I think about it i'm sure that's just from what was visible from where I logged in, since I didn't move around at all.

[fml-client-latest.zip](https://github.com/Ivorforce/Psychedelicraft/files/156051/fml-client-latest.zip)

**Comment by Ivorforce** [comment]
Yeah, that looks pretty much like some mod hooks its shaders in and then disables it shortly after, leaving Psychedelicraft's shaders disabled.
